src/libstd/thread/mod.rs

```diff
@@ -482,7 +484,10 @@ impl Builder {
         };
 
         Ok(JoinHandle(JoinInner {
-            native: Some(imp::Thread::new(stack_size, Box::new(main))?),
+            native: Some(imp::Thread::new(
+                stack_size,
+                mem::transmute::<Box<dyn FnBox()>, Box<dyn FnBox()>>(Box::new(main))
```

Comment:
Oh I thought this was accidental? How come this is necessary?

Reply:
The implementation in `sys` needs to be able to store the `Box<dyn FnBox() + 'static>` in a queue in a [static variable](https://github.com/rust-lang/rust/pull/56066/files#diff-6346734096cfe60a79cd965a5355f629R47). #55043 changed this from `'static` to `'a`. Rationale for this specific change is in https://github.com/rust-lang/rust/issues/55132#issuecomment-435164525 and following comments.

Reply:
Could lifetime annotations be added to the transmute parameters and could a comment be added as well?

Reply:
Done